src/securemul.rs

```diff
@@ -49,10 +50,11 @@ impl<F: Field> SecureMul<F> {
     /// ## Errors
     /// Lots of things may go wrong here, from timeouts to bad output. They will be signalled
     /// back via the error response
-    pub async fn execute<R: Ring>(
+    pub async fn execute<R: CommunicationGateway>(
         self,
         ctx: &ProtocolContext<'_, R>,
     ) -> Result<ReplicatedSecretSharing<F>, BoxError> {
+        let mut helper_ring = ctx.helper_ring.ring_channel(ProtocolId::from(self.index));
         // generate shared randomness.
         let (s0, s1) = ctx.participant.generate_fields(self.index);
```

Comment:
Thanks @martinthomson I was looking at #68 and I think I got the idea here, but want to clarify my understanding of it. Ben's use case in #63 for modulus conversion requires `256` multiplications to be performed (2 per bit). Does that mean that we would generate a participant with `256` spaces and conversion protocol will do the following (just a sketch to demonstrate the idea)

```rust
   for bit in 0..u128::BITS {
         // participant needs to be renamed to `prss_space`
         let ctx = ProtocolContext { participant: &participant[bit], share_a: ..., share_b: ... }
         securemul.execute(ctx)


   } 
    
```

Reply:
The modulus conversion would occur within a single space, one that is reserved for that purpose.

Thinking holistically, a complete run will involve multiple multiplications.  Each multiplication can be assigned a PRSS space.

(If you don't know how many you need, then make a large space and pick spaces sequentially as you go through the code.)

Reply:
in this case , would we give modulus conversion a participant with `N` spaces so it can borrow one for itself and leave N-1 for secure multiplications or we need another struct capable of producing `Participant` instances with reserved space (similar to `setup` but without public keys)?

Reply:
If the multiplications can be given indices from the same space, then you might reasonably use a single space for those multiplications.  It's going to depend on how you drive the multiplications, I think.

Reply:
Is my understanding correct that we would want `ProtocolId` to be some sort of composite id that includes both "step" and "index". So the space (step) needs to be exposed to `SecureMul` somehow so it can construct it from these two pieces